frontend/views/containers/contributions/IncomeDetails.vue

```diff
@@ -0,0 +1,239 @@
+import ModalBaseTemplate from '@components/Modal/ModalBaseTemplate.vue'
+import TextWho from '@components/TextWho.vue'
+import GroupPledgesGraph from '../GroupPledgesGraph.vue'
+import L from '@view-utils/translations.js'
+
+export default {
+  name: 'IncomeDetails',
+  mixins: [validationMixin],
+  components: {
+    ModalBaseTemplate,
+    InputAmount,
+    TextWho,
+    Tooltip,
+    PaymentMethods,
+    GroupPledgesGraph
+  },
+  data () {
+    return {
+      ephemeral: {
+        formFeedbackMsg: {}
+      },
+      form: {
+        incomeDetailsKey: null,
+        amount: null
+      },
+      fakeStore: {
+        incomeDetailsKey: 'pledgeAmount', // incomeAmount || pledgeAmount
+        amount: 100
+      }
+    }
+  },
+  computed: {
+    ...mapGetters([
+      'groupSettings',
+      'memberProfile',
+      'groupMincomeFormatted'
+    ]),
+    memberGroupProfile () {
+      const profile = this.memberProfile(this.$store.state.loggedIn.username) || {}
+      return profile.groupProfile || {}
+    },
+    needsIncome () {
+      return this.form.incomeDetailsKey === 'incomeAmount'
+    },
+    canPledge () {
+      return this.form.incomeDetailsKey === 'pledgeAmount'
+    }
+  },
+  created () {
+    const incomeDetailsKey = this.memberGroupProfile.incomeDetailsKey
+    if (incomeDetailsKey) {
+      this.form.incomeDetailsKey = incomeDetailsKey
+      this.form.amount = this.memberGroupProfile[incomeDetailsKey]
+    }
+  },
+  methods: {
+    resetAmount () {
+      this.form.amount = this.form.incomeDetailsKey === this.memberGroupProfile.incomeDetailsKey ? this.memberGroupProfile[this.memberGroupProfile.incomeDetailsKey] : ''
+      this.$v.form.$reset()
+    },
+    closeModal () {
+      this.$refs.modal.close()
+    },
+    async submit () {
+      if (this.$v.form.$invalid) {
+        alert(L('Invalid payment info'))
+        return
+      }
+      try {
+        const incomeDetailsKey = this.form.incomeDetailsKey
+        const groupProfileUpdate = await sbp('gi.contracts/group/groupProfileUpdate/create',
+          {
+            incomeDetailsKey,
+            [incomeDetailsKey]: +this.form.amount
+          },
+          this.$store.state.currentGroupId
+        )
+        await sbp('backend/publishLogEntry', groupProfileUpdate)
+        this.ephemeral.formFeedbackMsg = {
+          text: L('Income details saved successfully!'),
+          class: 'has-text-success'
+        }
+      } catch (e) {
+        console.error('setPaymentInfo', e)
+        this.ephemeral.formFeedbackMsg = {
+          text: L(`Failed to update income details. Error: ${e.message}`),
+          class: 'has-text-danger'
+        }
+      }
+    }
+  },
+  validations: {
+    form: {
+      incomeDetailsKey: {
+        [L('This field is required')]: required
+      },
+      amount: {
+        [L('This field is required')]: required,
+        [L('It seems your income is not lower than the group\'s mincome.')]: function (value) {
```

Comment:
I don't know if these backslashes will cause problems with the translation, can you replace all instances with double-quotes instead of single quotes so that you don't need `\'`?